### ISSUE TITLE
Keep docker image labels around after conversion

### ISSUE THREAD

**Issue by grepory** [appc/docker2aci#208]
It would appear that annotations are the equivalent to Docker labels: i.e. immutable image metadata. Is that the case, and if so, can we support keeping image labels around as annotations after conversion? Or as labels?

**Comment by lucab** [comment]
docker2aci is already adding those annotation while converting (perhaps it should be better documented):
```
  "annotations": [
    {
      "name": "created",
      "value": "2016-09-23T18:08:51Z"
    },
    {
      "name": "appc.io/docker/registryurl",
      "value": "registry-1.docker.io"
    },
    {
      "name": "appc.io/docker/repository",
      "value": "library/debian"
    },
    {
      "name": "appc.io/docker/imageid",
      "value": "37c816ae4431cabacbd1cf9ef8b50f9945ebc47a9aaa26a315612edc52b12c32"
    },
    {
      "name": "appc.io/docker/parentimageid",
      "value": "d0c4f1eb7dc8f4dae2b45fe5c0cf4cfc70e5be85d933f5f5f4deb59f134fb520"
    },
    {
      "name": "appc.io/docker/cmd",
      "value": "[\"/bin/bash\"]"
    },
    {
      "name": "appc.io/docker/tag",
      "value": "latest"
    }
  ]
```

I think those annotations so far cover metadata that we found was needed by other tools. Do they cover your usecase too or do you need something else additionally?

**Comment by jonboulle** [comment]
@lucab he's referring specifically to keeping docker labels around, which
we should probably add.

Luca Bruno <notifications@github.com> schrieb am Mi., 19. Okt. 2016, 11:52:

**Comment by jonboulle** [comment]
Ref: https://docs.docker.com/engine/reference/builder/#label
https://docs.docker.com/engine/userguide/labels-custom-metadata/

**Comment by grepory** [comment]
I did indeed mean that very thing. We use the LABEL directive in our docker files to encode build-time information into the image. Then when the image is signed, so is that static metadata. This is pretty important to our ability to track what is deployed: who built it, when did they build it, what git rev was used to build it, etc.

**Comment by jonboulle** [comment]
Makes sense. I think we can just create appc.io/docker/labels annotations
and store them in there.

Greg Poirier <notifications@github.com> schrieb am Mi., 19. Okt. 2016,
18:48:

**Comment by grepory** [comment]
We have tooling around the label-schema.org stuff. Would it be horribly inappropriate to keep the current label namespace?

**Comment by jonboulle** [comment]
Do you mean copy the labels to annotations?

Greg Poirier <notifications@github.com> schrieb am Mi., 19. Okt. 2016,
19:11:

**Comment by grepory** [comment]
Yes, if annotations are the equivalent of docker labels. What is the difference between an appc annotation vs appc label?

**Comment by jonboulle** [comment]
appc labels are more like selectors, used for discovery. So there should be
a small set of well known labels used as "query parameters" when trying to
locate an image (analogous to tags, but immutable).

Annotations can be arbitrary and hence are pretty much equivalent to docker
labels.

Greg Poirier <notifications@github.com> schrieb am Do., 20. Okt. 2016,
00:28:

**Comment by grepory** [comment]
Assuming that appc annotations == docker labes.

We right now use:
```
FROM #{docker_image}
LABEL org.label-schema.schema-version="1.0"
LABEL org.label-schema.name="#{job_name}"
LABEL org.label-schema.build-date="#{Time.now.to_datetime.rfc3339}"
LABEL org.label-schema.vcs-ref="#{git_rev}"
LABEL org.label-schema.os-release="#{os_release(docker_image)}"
LABEL org.label-schema.vendor="Stripe"
```

Pardon the Ruby, but every image we create gets these labels.

From the exported image, we take all of the labels from the config file referenced in manifest.json and make those annotations in the resulting appc manifest without modifying the label identifiers (use the label-schema.org naming scheme instead of the /-delimited namespacing that coreos uses).

**Comment by grepory** [comment]
https://github.com/appc/docker2aci/pull/209

**Comment by lucab** [comment]
Closed by #223.
